server/src/RelayHttpServer.go

```diff
@@ -351,6 +367,18 @@ func keepAlive() {
 	log.Println("Done registering")
 }
 
+func shutdownOnRelayRemoved() {
+	removed, err := relay.IsRemoved()
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	if removed {
+		server.Close()
```

Comment:
better force the `ready`-state to false in case `removed==true`.
This way, we can still query the relay's status.